kpops/pipeline_generator/pipeline.py

```diff
@@ -160,6 +171,55 @@ def load_from_yaml(
         pipeline = cls(main_content, env_content, registry, config, handlers)
         return pipeline
 
+    def __generate_graph(self) -> None:
```

Comment:
would it make sense to move all this inside the `PipelineComponents` class?

Reply:
note to self: we could also consider renaming the classes as they have become a bit outdated in regards to their resposibilities
`PipelineComponents` → `Pipeline`
`Pipeline` → `PipelineParser`

Reply:
I mean, how do you see it? I think right now it is in the right place if we consider the parse_components function is inside the pipeline class. Or what do you think ?

Reply:
well that's the yaml → Pipeline parsing, but since the graph generation is only from `list[PipelineComponent]` → DAG this should be moved to `PipelineComponents` class imo. good indicator of this is the usage of `self.components.graph` everywhere